File: EX2/interfaces/src/main/java/EnvData.java
import java.io.Serializable;

public class EnvData implements Serializable {
	final String mTimestamp;
	final String mSensor;
	final int[] mValues;

	EnvData(String _timestamp, String _sensor, int[] _values) {
		mTimestamp = _timestamp;
		mSensor = _sensor;
		mValues = _values;
	}
}

File: EX3/lib/src/main/kotlin/classes/EnvData.kt
package classes

import java.io.Serializable
import javax.xml.bind.annotation.*

/**
 * Our Class that holds our Sensordata
 * and gets converted from/to xml
 */
@XmlRootElement(name = "env-data")
data class EnvData(
    /**
     * seconds timestamp
     */
    @field:XmlElement
    val mTimestamp: String,
    /**
     * the name of our sensor
     */
    @field:XmlElement
    val mSensor: String,
    /**
     * all the values that the Sensor holds
     */
    @field:XmlElementWrapper(name = "values")
    @field:XmlElement(name = "value")
    val mValues: IntArray
) : Serializable {
    companion object {
        const val serialVersionUID = 123L
    }

    constructor() : this("", "", intArrayOf())
}